### DISCUSSION TITLE
How to stop section of text being split over multiple pages (PDF)

### DISCUSSION THREAD

**Discussion by MWP** [quarto-dev/quarto-cli#11231] [Q&A]
### Description

Hi all,
My source document is markdown, the target is PDF. Using scrartcl.

Is there a method to makes sure a section of text (a few paragraphs, plus code blocks) are never split over separate pages?
Something like a \dontsplitstart and \dontsplitend ?

Thanks in advance!

**Comment by mcanouil** [comment]
No.

Where the text should go if it does not fit the page?
Note that even Word can't do that.

**Reply by MWP** [reply]
Obviously if it's too long to fit on a single page, it should flow onto the next.
But if it can fit on a single page, and would otherwise flow onto a new page due to content before it, then a page break should be placed before the section to make sure it doesn't.

**Reply by mcanouil** [reply]
As far as I know, no tools do that.
That's user editing.

So use `{{< pagebreak >}}` when you need to.
Edit the LaTeX file to add pagebreaks manually (`format: latex`) and compile manually.

**Comment by Steinthal** [comment]
Maybe the OP is talking about widows and orphans. 

On the other hand, code blocks are floats, so the positioning of floats could also be part of the question. In that case, package ["float"](https://ctan.org/pkg/float) could help: "The package also provides the H float modifier option of the obsolete [here](https://ctan.org/pkg/here) package. You can select this as automatic default with \floatplacement{figure}{H}."

Instead of the default ````\def\fps@figure{htbp}````, I have this in my "graphics.tex" partial:

````tex
\usepackage{float}
\floatplacement{figure}{$if(float-placement-figure)$$float-placement-figure$$else$H$endif$}
````

**Reply by mcanouil** [reply]
Indeed, thanks for sharing this.

Note that this won't cover all possible float environments and as with everything in LaTeX, additional specific tweaks might be required.
For such tweaks, Quarto discussion board is not the best place to look for them.

The trick shared won't push a section to the next page to allow all content in section to fit in the page (I don't think LaTeX is able at all to do that).
